### PR TITLE
[FEATURE] Add a HtmlScrubber class

### PR DESCRIPTION
**Pull request by oliverklee**
This class can remove parts of the HTML. (In the `Emogrifier` class, this
was solved with switches for the emogrification process.)

Fixes #651

### CODE REVIEW

**Comment by oliverklee** [comment]
I'll also appreciate feedback on the name of the class.

**Comment by JakeQZ** [comment]
> Other possibile nouns...

A few more: de-redundantifier; pruner; eliminator.

**Comment by JakeQZ** [comment]
> I'll also appreciate feedback on the name of the class.

I am thinking now that `HtmlPruner` may be a better name.  There is an analogous usage when talking about “pruning a database” – removing redundant or unwanted things – and one definition of “prune” I found is “to reduce especially by eliminating superfluous matter”, which seems very fitting.

**Comment by oliverklee** [comment]
`HtmlPruner` sounds good. I'd like to go with that.

File: tests/Unit/Emogrifier/HtmlProcessor/HtmlScrubberTest.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+     *
+     * @dataProvider displayNoneDataProvider
+     */
+    public function removeInvisibleNodesRemovesNodesWithDisplayNone($displayNone)
+    {
+        $subject = HtmlScrubber::fromHtml('<html><body><div style="' . $displayNone . '"></div></body></html>');
+
+        $subject->removeInvisibleNodes();
+
+        self::assertNotRegExp('/display:\\s*none;?/', $subject->render());
```
[the remark this answers is not in the document]
Comment (oliverklee) [reply tests/Unit/Emogrifier/HtmlProcessor/HtmlScrubberTest.php]: Oh, that's right. I completely missed that. I'll change that in a minute.

**Comment by oliverklee** [comment]
Repushed.

File: tests/Unit/Emogrifier/HtmlProcessor/HtmlPrunerTest.php [hunk 2]
```diff
@@ -78,10 +78,10 @@ public function displayNoneDataProvider()
      */
     public function removeInvisibleNodesRemovesNodesWithDisplayNone($displayNone)
     {
-        $subject = HtmlScrubber::fromHtml('<html><body><div style="' . $displayNone . '"></div></body></html>');
+        $subject = HtmlPruner::fromHtml('<html><body><div style="' . $displayNone . '"></div></body></html>');
 
         $subject->removeInvisibleNodes();
 
-        self::assertNotRegExp('/display:\\s*none;?/', $subject->render());
+        self::assertNotContains('<div>', $subject->render());
```
[the remark this answers is not in the document]
Comment (oliverklee) [reply tests/Unit/Emogrifier/HtmlProcessor/HtmlPrunerTest.php]: Darn, you're right. This happens when I code before having had a decent coffee.

**Pull request merged**